### PR TITLE
Fix: Harden MongoDB query and options handling (fixes #58)

### PR DESCRIPTION
**Pull request by taylortom**
### Fix
- Block dangerous MongoDB query operators (`$where`, `$function`, `$accumulator`) that allow arbitrary JS execution on the server
- Add allowlist for MongoDB driver options — only `collation`, `includeResultMetadata`, `limit`, `projection`, `returnCursor`, `returnDocument`, `skip`, `sort`, `upsert` are permitted
- Validate option value types (integers, booleans, plain objects, enums) before passing to the driver
- Return copies from parameter processing instead of mutating caller inputs

### Update
- Extract `processParams` utility that consolidates query validation, ObjectId conversion, `_id` stripping, and options parsing into a single function
- Add `assertSafeQuery` utility for recursive detection of blocked query operators
- Consolidate barrel file (`lib/utils.js`) with all utility exports
- Simplify `MongoDBModule` methods by replacing scattered inline processing with single `processParams` call

### Testing
- 16 tests for `assertSafeQuery` (safe operators, blocked operators, nesting, edge cases)
- 24 tests for `processParams` (query/data/options processing, immutability, allowlist, value validation)
- All 56 tests passing

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: lib/MongoDBModule.js [hunk 1]
```diff
@@ -169,15 +141,9 @@ class MongoDBModule extends AbstractModule {
...
    */
   async update (collectionName, query, data, options) {
-    const opts = Object.assign({ includeResultMetadata: false, returnDocument: 'after' }, options)
-    this.parseOptions(opts)
-    convertObjectIds(query)
-    convertObjectIds(data)
-    // MongoDB doesn't like the explicit setting of _id
-    delete data._id
-    if (data.$set) delete data.$set._id
+    const p = processParams({ query, data, options: { includeResultMetadata: false, returnDocument: 'after', ...options } })
```
[the remark this answers is not in the document]
Comment (taylortom) [reply lib/MongoDBModule.js:144]: Not a real issue — in ES2018+, spreading `undefined` or `null` in an object literal (`{ ...undefined }`) is valid and produces `{}`. No runtime throw on Node 24.

File: lib/MongoDBModule.js [hunk 2]
```diff
@@ -218,15 +179,9 @@ class MongoDBModule extends AbstractModule {
...
    */
   async replace (collectionName, query, data, options) {
-    const opts = Object.assign({ includeResultMetadata: false, returnDocument: 'after' }, options)
-    convertObjectIds(query)
-    convertObjectIds(data)
-    this.parseOptions(options)
-    // MongoDB doesn't like the explicit setting of _id
-    delete data._id
-    if (data.$set) delete data.$set._id
+    const p = processParams({ query, data, options: { includeResultMetadata: false, returnDocument: 'after', ...options } })
```
[the remark this answers is not in the document]
Comment (taylortom) [reply lib/MongoDBModule.js:182]: Not a real issue — in ES2018+, spreading `undefined` or `null` in an object literal (`{ ...undefined }`) is valid and produces `{}`. No runtime throw on Node 24.

File: lib/utils/processParams.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  if (data) {
+    result.data = { ...data }
+    convertObjectIds(result.data)
+    if (!options?.preserveId) {
+      delete result.data._id
+      if (result.data.$set) delete result.data.$set._id
+    }
+  }
```
[the remark this answers is not in the document]
Comment (taylortom) [reply lib/utils/processParams.js:56]: Fixed in 5a756da — switched to `structuredClone()` so nested objects are fully isolated before `convertObjectIds` runs.

File: lib/utils/processParams.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+ * @type {Object<string, function>}
+ */
+const OPTION_VALIDATORS = {
+  collation: v => isObject(v) ? v : undefined,
+  includeResultMetadata: v => typeof v === 'boolean' ? v : undefined,
+  limit: v => { const n = parseInt(v); return Number.isFinite(n) ? n : undefined },
+  projection: v => isObject(v) ? v : undefined,
+  returnCursor: v => typeof v === 'boolean' ? v : undefined,
+  returnDocument: v => v === 'before' || v === 'after' ? v : undefined,
+  skip: v => { const n = parseInt(v); return Number.isFinite(n) ? n : undefined },
```
[the remark this answers is not in the document]
Comment (taylortom) [reply lib/utils/processParams.js]: Fixed in 5a756da — replaced `parseInt` with `Number()` + `Number.isInteger()` so partial strings like `'10abc'` and floats like `'1.5'` are correctly rejected.
